src/core_plugins/kibana/server/lib/make_usage_collector.js

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { getUiSettingDefaults } from '../../ui_setting_defaults';
```

Comment:
@tsullivan if you could do a general review, especially of the stuff that's *not* related to the saved object API (since I have the others looking at it), I'd be thankful.

Reply:
This looks good to me! other than the naming, though. I'm not sure if this will some day be general purpose for other collectors to live alongside yours. But I think it would be more clear if the file was `kql_usage_collector` and the object being made and the function being exported named more clearly. Honestly I had some confusion when I first opened this up, because there is a different "kibana usage collector" that lives in [x-pack/monitoring](https://github.com/elastic/kibana/blob/master/x-pack/plugins/monitoring/server/kibana_monitoring/collectors/get_kibana_usage_collector.js) (for now, because of BWC, i.e. `formatForBulkUpload` which puts the data into the model that the upstream expects it to be)

Reply:
Ah so that's where that "kibana" telemetry document comes from? I asked about it in the issue I filed in the telemetry repo but I didn't hear back. Question for you then: do you think this data should go under that kibana namespace? It feels like it should since it's coming from the Kibana plugin. I don't know how feasible that is if the current collector is in x-pack, or if it's too late to change the mapping schema.

Reply:
> do you think this data should go under that kibana namespace?

Personally I don't have strong feelings one way or the other, but it's true that it's a bit late to try to change the model here. Maybe @pickypg can weigh in on that, since he's the one updating the telemetry cluster mappings.

If it's highly preferable to put this side-by-side with other basic Kibana stats, the most straightforward way of doing that would probably be to expand the existing Kibana collector functionality to include the KQL stats also. I really want to avoid any "after the fact" merging stats from different collectors into single namespaces, except for backwards compatibility.